pokemongo_bot/cell_workers/recycle_items.py

```diff
@@ -91,25 +152,34 @@ def work(self):
         worker_result = WorkerResult.SUCCESS
         if self.should_run():
 
-            if not (self.max_balls_keep is None):
-                worker_result = self.recycle_excess_category_max(self.max_balls_keep, [1,2,3,4])
-            if not (self.max_potions_keep is None):
-                worker_result = self.recycle_excess_category_max(self.max_potions_keep, [101,102,103,104])
-            if not (self.max_berries_keep is None):
-                worker_result = self.recycle_excess_category_max(self.max_berries_keep, [701,702,703,704,705])
-            if not (self.max_revives_keep is None):
-                worker_result = self.recycle_excess_category_max(self.max_revives_keep, [201,202])
-
-            inventory.refresh_inventory()
-
             for item_in_inventory in inventory.items().all():
-
                 if self.item_should_be_recycled(item_in_inventory):
                     # Make the bot appears more human
                     action_delay(self.recycle_wait_min, self.recycle_wait_max)
                     # If at any recycling process call we got an error, we consider that the result of this task is error too.
                     if ItemRecycler(self.bot, item_in_inventory, self.get_amount_to_recycle(item_in_inventory)).work() == WorkerResult.ERROR:
                         worker_result = WorkerResult.ERROR
+
+            if not (self.max_balls_keep is None):
```

Comment:
Why putting this after the recycling based on item amount ? The bot is discarding high balls before looking for lower balls to discard instead of highers one. You’re making the recycling based on type useless for people using both of them (with a well configured task).